src/blockdata/block.rs

```diff
@@ -455,6 +475,19 @@ mod tests {
         assert_eq!(xfield.unwrap(), XField::AggregatePublicKey(pk));
     }
 
+    #[test]
+    fn xfield_max_block_size_test() {
+        let bytes = Vec::from_hex("0200010000").unwrap();
+        let decode: XField = deserialize(&bytes).unwrap();
+        assert_eq!(serialize(&decode), bytes);
+
+        let maxblocksize:u32 = 0x100;
+        assert_eq!(decode, XField::MaxBlockSize(maxblocksize));
+
+        let xfield = XField::from_str("0200010000");
+        assert_eq!(xfield.unwrap(), XField::MaxBlockSize(maxblocksize));
```

Comment:
Please add test cases for normal values as well as boundary and abnormal values.